src/pages/search/index.tsx

```diff
@@ -1,131 +1,80 @@
-import { useRouter } from "next/router";
 import { useEffect, useState } from "react";
-import { useQuery } from "react-query";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+
+import { Icon } from "@iconify/react";
+
 import Header from "../../components/common/header";
 import ProductItem from "../../components/main/product-item";
-import useToast from "../../hooks/useToast";
-import { ProductDataMin } from "../../common/types/data.types";
-import { apiGet } from "../../service/request";
 import Toast from "src/components/common/ui/toast";
+import SearchWords from "src/components/search/search-words";
 
-interface KeywordInterface {
-  id: number;
-  tag: string;
-}
+import useToast from "../../hooks/useToast";
+import useRecentWord from "src/hooks/useRecentWord";
+import { ProductDataMin } from "../../common/types/data.types";
+import { useSearchProduct } from "src/service/query/search";
 
 const Search: NextPage = () => {
-  const session = useSession();
-  const router = useRouter();
-
-  const { setToast, showToast, toastController } = useToast();
-
-  const [keywords, setKeywords] = useState<string[]>([]);
-
   const [inputValue, setInputValue] = useState<string>("");
-
   const [searchData, setSearchData] = useState<ProductDataMin[]>([]);
+  const [isSearch, setIsSearch] = useState<boolean>(false);
 
-  const [matchedKeywords, setMatchedKeywords] = useState<string[]>([]);
-
-  const [focus, setFocus] = useState<boolean>(false);
-
-  // ------------------------------------------------------------------------------------
-  const CACHE_KEY = "recentSearches";
-  const EXPIRATION_TIME = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
-
-  const [searches, setSearches] = useState<string[]>([]);
-
-  useEffect(() => {
-    if (session.status !== "unauthenticated") {
-      caches.open(`my-cache-${session.data?.user?.name}`).then(cache => {
-        cache.match(CACHE_KEY).then(response => {
-          if (response) {
-            response.json().then(data => {
-              setSearches(data);
-            });
-          }
-        });
-      });
-    }
-  }, []);
-
-  useEffect(() => {
-    if (session.status !== "unauthenticated") {
-      caches.open(`my-cache-${session.data?.user?.name}`).then(cache => {
-        const expirationDate = new Date().getTime() + EXPIRATION_TIME;
-        const cacheHeaders = new Headers({
-          "Content-Type": "application/json",
-          Expires: new Date(expirationDate).toUTCString(),
-        });
-        const cacheOptions = {
-          headers: cacheHeaders,
-        };
-        const uniqueSearches = [...new Set(searches)]; // 중복 제거
-        cache.put(new Request(CACHE_KEY), new Response(JSON.stringify(uniqueSearches), cacheOptions));
-      });
-    }
-  }, [searches]);
+  const router = useRouter();
 
-  // ------------------------------------------------------------------------------------
+  const enteredWord = router.query.word as string;
```

Comment:
```suggestion
  const enteredWord = router.query.word || '';
```
위처럼 처리하면 타입 단언 제거해도 되지 않나요!?

Reply:
query.word가 string | string[]으로 유니온 타입을 가지는데 enteredWord를 전달받는 함수의 타입은 string으로 고정되어있어 생기는 타입오류를 막기위해 상위에서 string타입을 보장함이 필요합니다~! 타입단언이 가장 명확하고 단순하게 타입을 보장할 수 있어서 사용한 부분입니다.